src/python/py/models/builder.py

```diff
@@ -2491,6 +2664,11 @@ def create_model(model_name, input_path, output_dir, precision, execution_provid
             onnx_model = Phi3Mini4KModel(config, io_dtype, precision, execution_provider, cache_dir, extra_options)
         elif config.architectures[0] == "Phi3ForCausalLM" and config.max_position_embeddings == 131072:
             onnx_model = Phi3Mini128KModel(config, io_dtype, precision, execution_provider, cache_dir, extra_options)
+        elif config.architectures[0] == "PhiMoEForCausalLM" and config.max_position_embeddings == 131072:
+            print("WARNING: This model only works for CUDA currently because `MoE` is only supported for CUDA in ONNX Runtime. Setting `--execution_provider cuda` by default.")
+            print("WARNING: This model currently only supports quantized version.")
+            execution_provider = "cuda"
```

Comment:
Should we set `precision = "int4"` as well since that's the only option?

Reply:
fp16 should also work but yeah let me hardcode it to int4 for less ambiguity